### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
Replacing https://github.com/kubevirt/ssp-operator/pull/836.
The #836 requires some more changes, which dependabot cannot do

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by akrejcir** [comment]
Looks good. Let's see if CI passes.

/lgtm

File: go.mod [hunk 1]
```diff
@@ -5,128 +5,131 @@ go 1.20
...
 	github.com/onsi/gomega v1.30.0
 	github.com/openshift/api v0.0.0-20231130063041-be20875a1373 // release-4.15
 	github.com/openshift/custom-resource-status v1.1.2
 	github.com/openshift/library-go v0.0.0-20231129101644-c7f15dcc10f5 // release-4.15
-	github.com/operator-framework/api v0.20.0
-	github.com/operator-framework/operator-lib v0.11.1-0.20231020142438-152ee1fb7f83 // TODO: Use v0.12.0 when it's released
-	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.69.1
-	github.com/prometheus/client_golang v1.17.0
+	github.com/operator-framework/api v0.21.0
+	github.com/operator-framework/operator-lib v0.12.0 // TODO: Use v0.12.0 when it's released
```
Comment (akrejcir) [inline go.mod:17]: Can you remove the comment?

Comment (ksimon1) [reply go.mod:17]: done

**Comment by ksimon1** [comment]
closed in favor of https://github.com/kubevirt/ssp-operator/pull/841.

**Pull request closed without merging**